xapian-core/common/parseint.h

```diff
@@ -0,0 +1,44 @@
+/** @file parseint.h
+ * @brief Parse unsigned type from string and check for trailing characters.
+ */
+/* Copyright (C) 2018 Olly Betts
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+#ifndef XAPIAN_INCLUDED_PARSEINT_H
+#define XAPIAN_INCLUDED_PARSEINT_H
+
+#ifndef PACKAGE
+# error config.h must be included first in each C++ source file
+#endif
```

Comment:
I don't think this check is needed here.

Some headers check this (which I guess is where you copied it from), but that's because they rely on configure-probe results from `config.h`, e.g.: in `closefrom.h` we test:

```
#ifdef HAVE_CLOSEFROM
```

If `closefrom.h` is included from a C++ source which doesn't include `config.h` first, then `HAVE_CLOSEFROM` won't be defined but we probably won't get a compile error but will in this case just end up using a less efficient fall-back version of the code.  More generally it can cause hard to debug problems if different parts of the code think a feature is and isn't available.

We could explicitly `#include <config.h>` from headers as well as source files to avoid this problem, but ISTR that's not recommended.  Not sure of the details though.

Reply:
thanks i will fix that.